deploy.sh

```diff
@@ -23,6 +23,13 @@ elif [ "$1" == "build" ]; then
 		exit 1
 	fi
 	echo "build finished"
+elif [ "$1" == "test" ]; then
+	./deploy.sh build
+	rm -rf coverage
+	mkdir coverage
+	./out/run_all_tests
+	gcovr -r . --html --html-details -o coverage/coverage-report.html -e Test/
+	echo "Open ${PWD}/coverage/coverage-report.html to view coverage report"
```

Comment:
Optional, but would suggest amending CodeCov instructions + an example to the `Test` README

Reply:
> 
> 
> Running `./deploy.sh format` generates a diff on `badgerloop.cpp` + most/all of the autocoded files. Maybe outside the scope of this PR, but there may be an issue with autocoded templates not being formatted properly. Maybe worth pre-formatting them. Unsure why there's a diff with `badgerloop.cpp`

Yeah I noticed that but decided it's outside the scope of this PR. Feel free to raise an issue.